### ISSUE TITLE
FIDO2 WebAuthn tokens are not created as residentKeys, causing them to fallback to U2F tokens

### ISSUE THREAD

**Issue by Braddux** [bitwarden/clients#5382]
### Steps To Reproduce

The following was tested in Chrome and Brave on linux as well as on a Chromebook and using Android devices as the FIDO2 device

1. Visit https://vault.bitwarden.com/#/login
2. Login
3. Profile -> Account Settings
4. Security
5. Two-step login
6. WebAuthn -> manage
7. *Master Password*
8. Enter a Name and click read key
9. Either select a pared android device or scan the QR code
10. Approve and authenticate on the Android device
11. Select save in the Chrome browser

Attempt to login to Bitwarden on other devices and note that you can only use the original Android device as the security key not other devices like FIDO2 WebAuthn (passkeys) are supposed to work and be syncronized via Google password manager. They don't appear in passwords.google.com.  They do sync on iOS devices.

### Expected Result

The key created this way should be a FIDO2 WebAuthn (passkey) secret, but it is falling back to a legacy U2F token. This prevents it from being synced via Google's password manager like it is on iOS via iCloud.

### Actual Result

The secret is device locked and with Bitwarden's limit of 5 devices this is a limiting factor. It is also inconsistent with how Apple and iCloud work.

### Additional Context

According to https://passkeys.dev/docs/use-cases/bootstrapping/#opting-the-user-into-passkeys this should be fixed by adding the key-value pair residentKey="required" when creating the key.

### Operating System

Linux, Android

### Web Browser

Chrome, Brave

### Build Version

2023.4.0

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by Krychaz** [comment]
Hello there,

It does indeed look like an issue. I have vetted this issue and passed it to our team.

**Comment by Enrico204** [comment]
If I understand this correctly, requiring the resident key means that end user devices should have enough space to hold as much as resident keys as the number of services the user is using. This is not something that a hardware key (e.g., nitrokeys, yubikeys) supports, as the storage is very limited. Maybe, there should be an option to choose from resident and non-resident keys, or use `rk=discouraged`.

See https://fy.blackhats.net.au/blog/2023-02-02-how-hype-will-turn-your-security-key-into-junk/

**Comment by TheCrazyMax** [comment]
> If I understand this correctly, requiring the resident key means that end user devices should have enough space to hold as much as resident keys as the number of services the user is using. This is not something that a hardware key (e.g., nitrokeys, yubikeys) supports, as the storage is very limited. Maybe, there should be an option to choose from resident and non-resident keys, or use `rk=discouraged`.
> 
> See https://fy.blackhats.net.au/blog/2023-02-02-how-hype-will-turn-your-security-key-into-junk/

Exactly, I had the same article in mind. For me this is not an issue, It's more the definition that got it wrong initially, and it stayed like that in people's mind

**Comment by closebot-bw** [comment]
⚠️ **Stale Issue Notice**

This issue has been automatically marked as stale due to inactivity. It will be closed in **2 weeks** (October 15, 2025) if no further activity occurs.

If this issue is still relevant and you would like to keep it open, please:
- Comment on this issue to show continued interest
- Provide any additional information or updates
- Confirm that the issue still exists in the latest version

Thank you for your contribution to this project! 🙏

**Comment by closebot-bw** [comment]
🔒 **Issue Closed Due to Inactivity**

This issue has been automatically closed due to lack of activity for an extended period. We periodically review and close inactive issues to help maintain our issue tracker and focus on current priorities.

**If this issue is still relevant:**
- Please create a new issue with updated information
- Include steps to reproduce the problem if it's a bug report
- Mention if this issue still occurs in the latest version

Thank you for your contribution to this project. Your feedback helps us improve! 🙏
